include/dt.h

```diff
@@ -0,0 +1,62 @@
+/***************************************************************************//**
+ *   @file   dt.h
+ *   @brief  Linux device tree like functionality.
+ *   @author Darius Berghe (darius.berghe@analog.com)
+********************************************************************************
+ * Copyright 2021(c) Analog Devices, Inc.
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *  - Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#ifndef _DT_H_
+#define _DT_H_
+#include <stdint.h>
+
+struct dt_property {
+	const char *name;
+	const int value; // contains the number of elements in array when array is given
+	void *array;
+};
+
+struct dt_properties {
+	struct dt_property *table;
```

Comment:
Maybe some macro for declaring dt_properties will be useful

Reply:
Can you give an example ?

Reply:
```
struct dt_property link_tx_ps_init[] = {
DT_ARRAY("logical-lane-mapping", {0, 2, 7, 6, 1, 5, 4, 3}),
DT_U32("link-mode", 9),
DT_U32("subclass", 1)
...
};
```
It is just an idea, I don't know if it is better

Reply:
Interesting idea but probably it'd be something like DT_ARRAY and DT_VALUE (value is stored as signed int) only. Also, you'd need to explicitly cast the type of the array, otherwise it won't compile.
`DT_ARRAY("logical-lane-mapping", (uint16_t []){0, 2, 7, 6, 1, 5, 4, 3}),`
So again, I'd implement this if it'd be of any help, but seems to only add verbosity to code.